### PR TITLE
Plan domain swap: New experiment beta

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

**For more context, also check pbmo2S-oh-p2.**

* This A/B test is part of the beta program for the new experiments platform.
* This test is a repeat of https://github.com/Automattic/wp-calypso/pull/41982 with a small change that swaps the domain and plan step in the signup flow.
* The following changes are introduced:
    - If a user in the variant group selects a paid plan, then show the domain step with text "Free with your plan".
    - If a user in the variant group selects a free plan, then show only the domain price without any "free with a paid plan text". Take them to checkout directly with only a domain in cart.

**Paid plan selected -> Show domain step with striked out price and 'Free with your plan' text**

Screenshot 2020-07-01 at 6 53 19 PM

**Free plan selected -> show full domain price -> directly take to checkout with only domain in cart**

Screenshot 2020-07-01 at 6 50 25 PM

Screenshot 2020-07-01 at 6 50 57 PM

#### Testing instructions

**Scenario 1**

1. Proxy through a US IP address and begin signup at /start.If in `control`, verify that the `onboarding` flow works well. If in variant group, then verify the following:

- Selecting a free plan shows the domain step with domain price and renewal price listed; you should see a notice message about domain forwarding; selecting a domain should take you to checkout page with only a domain in cart.

- Selecting a paid plan shows the domain step with striked out price and with text "Free with your plan". 

**Scenario 2**

* Proxy through a non-US IP and verify that you don't get assigned to the test.
 

**Scenario 3**

* From a US IP address, verify that you don't get assigned to the test when beginning signup from the following flows:
/start/personal, /start/premium, /start/business, /start/ecommerce.
* From a US IP address, verify that you're not assigned to the test if using wpcom in a non-EN locale.

### CODE REVIEW

**Comment by withinboredom** [comment]
So, I wouldn't handle the experiment here, but the next level up where `goToNextStep` is passed as a prop.

```
const props = {something: "here", otherProp: "value", etc: "etc"};
let alreadyStepped = false;
return (<Experiment>
  <DefaultVariation>
    <UserStep {...props} goToNextStep={() => goToNextStep()} />
  </DefaultVariation>
  <Variation name="treatment">
    <UserStep {...props} goToNextStep={() => {
      if(!alreadyStepped) {
        goToNextStep('onboarding-plan-first');
        alreadyStepped = true;
      } else {
        goToNextStep();
      }
    } />
  </Variation>
</Experiment>)
```

I'm not sure where this component is mounted at, so it's hard to tell how viable it is. There's also a problem with this implementation in that going back and then forward would give them the default variation, but I think that it can be better handled in the `goToNextStep` implementation, where ever it is.

File: client/components/domains/domain-product-price/index.jsx [hunk 1]
```diff
@@ -47,13 +52,29 @@ class DomainProductPrice extends React.Component {
...
-					message = translate( 'Registration fee: {{del}}%(cost)s{{/del}} {{span}}Free{{/span}}', {
-						args: { cost: this.props.price },
-						components: {
-							del: <del />,
-							span: <span className="domain-product-price__free-price" />,
-						},
-					} );
+					if ( selectedPaidPlanInSwapFlow ) {
+						message = translate(
+							'Registration fee: {{del}}%(cost)s{{/del}} {{span}}Free with your plan{{/span}}',
```
Comment (a8ci18n) [inline client/components/domains/domain-product-price/index.jsx:57]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Registration fee: {{del}}%(cost)s{{/del}} {{span}}Free with paid plan{{/span}}' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/swap-test-new-experiment-beta)

File: client/signup/controller.js [hunk 2]
```diff
@@ -263,6 +265,45 @@ export default {
 			context.store.dispatch( setSelectedSiteId( null ) );
 		}
 
+		if ( flowName === 'onboarding' || flowName === 'onboarding-plan-first' ) {
+			context.primary = (
+				<Experiment name="domain_step_plan_step_swap">
```
Comment (aaronyan) [inline client/signup/controller.js:270]: I started the experiment without this PR being merged, so we'll have to change this experiment name. How about `signup_domain_plan_step_swap`?

**Comment by jaswrks** [comment]
@aneeshd16 writes...
> This seems to have conflicts. I can help in testing this :slightly_smiling_face:

**Comment by aaronyan** [comment]
Screenshot for what the Spanish translations will look like when setting my browser language to `es`. Thank you for the translations @victorespigares!

**Spanish version**
Screen Shot 2020-07-30 at 1 06 15 PM

**Compared to the English version**
Screen Shot 2020-07-30 at 1 03 38 PM

**Comment by niranjan-uma-shankar** [comment]
@aaronyan I updated the PR with the new test name and variation - `signup_domain_plan_step_swap` and `variant_plans_first`. 

After I assigned myself to the variant in the MC tool for experiments for Experiment 91, I visited http://calypso.localhost:3000/start/user. The state variable `getState().experiments.variations` lists two tests - `signup_domain_plan_step_swap` and `domain_step_plan_step_swap`. Is this something to be fixed on the experiment definition in the tool?

**Comment by aaronyan** [comment]
Thanks for making those adjustments and for testing @niranjan-uma-shankar!

> The state variable getState().experiments.variations lists two tests - signup_domain_plan_step_swap and domain_step_plan_step_swap. Is this something to be fixed on the experiment definition in the tool?

Great question, and yeah this is on our end. I've disabled `domain_step_plan_step_swap` for now to clear up any confusion. :)

File: client/signup/controller.js [hunk 3]
```diff
@@ -246,6 +248,45 @@ export default {
...
```
Comment (aaronyan) [inline client/signup/controller.js:253]:
```suggestion
				<Experiment name="signup_domain_plan_step_swap">
```
I think the experiment name needs to be updated!

**Comment by aaronyan** [comment]
Tested locally (with this [commit suggestion](https://github.com/Automattic/wp-calypso/pull/43675#discussion_r464529538) and adjusting the assigned variation on the MC tool for experiment 91), and looks good to me! I think we're ready to merge in time to start the experiment on August 5 UTC.

**Pull request merged**